### PR TITLE
[navigation] populate incoming speed camera place page

### LINKED ISSUE #1657: Incoming speed camera selection empty on Android
In navigation mode, when a speed camera turns red because it is incoming, selecting it brings an empty popup. Other cameras work as expected and selecting them shows correct information.

| Incoming camera selected (the red one) | Camera on the other way selected (the black one) |
|-|-|
| Screenshot_20211204-222031_Organic_Maps | Screenshot_20211204-222040_Organic_Maps |

- OS: Android 9
- Organic Maps version: 2021.11.04-2-F-Droid
- Data Version: 211022

### PR DESCRIPTION
**Pull request by arnaudvergnet**
Adds information in the place page for incoming speed cameras (the ones shown in red).

For the moment, the place page shows the camera position and the title is the max speed if any.

I need help for this one as I have trouble retrieving the speed camera information. The `SpeedCameraMark` object does not contain a lot of information, and I was not able to retrieve the POI/Feature associated.

Here is what it looks like for the moment:
Screenshot_1656782339

But here is how regular speed cameras look like:
Screenshot_1656782706

I would like to retrieve the information from the regular speed camera and integrate it in the `SpeedCameraMark` but I am lost on how to proceed. Could please help me understand where to look?

Closes #1657

### CODE REVIEW

**Comment by biodranik** [comment]
> I would like to retrieve the information from the regular speed camera and integrate it in the SpeedCameraMark

@vng is it possible?

**Comment by biodranik** [comment]
@arnaudvergnet is this UX really needed? Drivers usually don't have time and the opportunity to tap the screen while driving. All camera info should be drawn on the screen automatically or announced via TTS.

**Comment by arnaudvergnet** [comment]
> @arnaudvergnet is this UX really needed? Drivers usually don't have time and the opportunity to tap the screen while driving. All camera info should be drawn on the screen automatically or announced via TTS.

The target is not the driver but the copilot. Everything you click in OM shows the place page, it would be strange if those speed cameras did not. The infoation people would be certainly looking for is the distance to the camera.

**Comment by biodranik** [comment]
Generally, yes, info may be displayed.
But cameras are displayed in navigation mode only at the moment, the combination of co-pilot + navigation mode is a rare one, so the priority of that task is definitely not the highest one.

**Comment by arnaudvergnet** [comment]
Considering how bad the TTS is on phones without Google, I would not say having a copilot is a very rare case. If it is too much work to show the camera info, a quick workaround would be to prevent the empty place page from showing.

**Comment by biodranik** [comment]
Yeah, showing at least something, without much effort and risks to break something, is good enough.

**Comment by rtsisyk** [comment]
I didn't get idea of this patch, but it would be nice to display the speed limit on the top of navigation screen.

**Comment by arnaudvergnet** [comment]
> I didn't get idea of this patch, but it would be nice to display the speed limit on the top of navigation screen.

In current version of OM when you select an incoming speed camera, it opens an empty place page. This makes the app feel unfinished and impacts user experience. This PR aims to at least fill some information in the place page. If filling information is too difficult, the place page should not be shown.

**Review by vng — APPROVED** [review]

**Review by rtsisyk — APPROVED** [review]

File: map/framework.cpp [hunk 1]
```diff
@@ -732,6 +732,19 @@ void Framework::FillRouteMarkInfo(RouteMarkPoint const & rmp, place_page::Info &
   info.SetIntermediateIndex(rmp.GetIntermediateIndex());
 }
 
+void Framework::FillSpeedCameraMarkInfo(SpeedCameraMark const & speedCameraMark, place_page::Info & info) const
+{
+  info.SetCanEditOrAdd(false);
+  info.SetMercator(speedCameraMark.GetPivot());
+
+  // Primary text is a speed limit, if any.
+  auto const titles = speedCameraMark.GetTitleDecl();
```
Comment (biodranik) [inline map/framework.cpp:741]: Are measurement units displayed? They are necessary.

**Review by biodranik — CHANGES_REQUESTED** [review]

**Comment by biodranik** [comment]
Speed Camera type should also be visible.

File: map/framework.cpp (hunk 1 above)
Comment (arnaudvergnet) [reply map/framework.cpp:741]: I would love to display those but I don't know how to retrieve them

Comment (biodranik) [reply map/framework.cpp:741]: @vng knows )

**Comment by arnaudvergnet** [comment]
> Speed Camera type should also be visible.

I have trouble understanding how the place page works in the cpp side. Seems like incoming (red) speed cameras are not handled the same way as regular (black) speed cameras. They are "user marks" and not features so I do not know how to retrieve the necessary data.

**Comment by vng** [comment]
PTAL

**Comment by vng** [comment]
I'm unable to fully test it on mobiles.

**Comment by arnaudvergnet** [comment]
Works perfectly thanks!

Screenshot_1662969831

**Comment by vng** [comment]
@biodranik PTAL

**Review by biodranik — APPROVED** [review]
Can be merged if works properly on iOS too.

**Pull request merged**